scene/gui/line_edit.cpp

```diff
@@ -255,11 +255,36 @@ void LineEdit::_gui_input(Ref<InputEvent> p_event) {
 				selection.creating = true;
 
 			} else {
-				if (b->is_doubleclick() && selecting_enabled) {
-					selection.enabled = true;
-					selection.begin = 0;
-					selection.end = text.length();
-					selection.doubleclick = true;
+				if (selecting_enabled) {
+					if (!b->is_doubleclick() && (OS::get_singleton()->get_ticks_msec() - selection.last_dblclk) < 600) {
+						// Triple-click select all.
+						selection.enabled = true;
+						selection.begin = 0;
+						selection.end = text.length();
+						selection.doubleclick = true;
+						selection.last_dblclk = 0;
+					} else if (b->is_doubleclick()) {
+						// Double-click select word.
+						selection.enabled = true;
+
+						int beg = cursor_pos;
+						int end = beg;
+						bool symbol = beg < text.length() && is_symbol(text[beg]);
+
+						while (beg > 0 && text[beg - 1] > 32 && (symbol == is_symbol(text[beg - 1]))) {
+							beg--;
+						}
+						while (end < text.length() && text[end + 1] > 32 && (symbol == is_symbol(text[end + 1]))) {
+							end++;
+						}
+						if (end < text.length()) {
+							end += 1;
+						}
```

Comment:
This should use `Vector<Vector2i> words = TS->shaped_text_get_word_breaks(text_rid);` instead of `is_symbol` and `> 32` to correctly detect words in the languages which do not use spaces, see `_move_cursor_left`, `_move_cursor_right` functions for example.

Reply:
But the current approach is fine for 3.2 version.

Reply:
Nice, I will update this PR doing that on `LineEdit` and `TextEdit` and open a new PR with this code for 3.2.

Reply:
Done